lib/class.nsync.php

```diff
@@ -43,65 +57,209 @@ public static function add_network_sites() {
 			$active = self::$settings['active'];
 		} 
 		
-		?>
-		<div id="select-site">
-		<?php
-		if( is_array($active) ):
-			?>
-				<p>Selected sites are able to publish content.</p>
-				<?php
-			foreach( $active as $active_site_id ):
-				$details = get_blog_details( array( 'blog_id' => $active_site_id ) );
-				
-				/*
-				$archived = ( $details->archived ? '<span> archived</span> ': '' ); 
-				$mature = ( $details->mature ? '<span> mature</span> ': '' );
-				$spam	= ( $details->spam ? '<span> spam</span> ': '' );
-				<?php echo $archived; ?><?php echo $spam; ?><?php echo $mature; ?>
-				*/
-				if( ( ! $details->archived && ! $details->spam && ! $details->deleted ) && $active_site_id != $current_blog_id ): ?>
-					<label>
-						<input name='nsync_options[active][]' type='checkbox' checked="checked" value='<?php echo esc_attr( $active_site_id ); ?>' />
-						<?php echo $details->blogname;?> <small><?php echo $details->siteurl;?> </small>
-					</label><br />
-					<?php
-				endif;
-			endforeach;
-			
-		else:
-		?>
-		<p>There are currently no external sites that are able to publish content on your site.</p>
-		<?php
-		endif;
-		?>
-		</div>
-		<?php 
+		// The admin tab navigation.
+		echo "<div id='nsync-nav'><div id='current'> <p>Current</p> </div> <div id='add'> <p>Add Sites</p> </div><div id='general'> <p>General</p> </div></div>";
+		
+		// add current publishing sites module.
+		echo "<div class='current nsync'>";
+		echo self::createPublishingList( $active );
+		echo "</div>";
 		
-		if( $current_user->ID ):
+		// add modules for adding sites to push list.
+		echo "<div class='add nsync'>";
+		echo self::createSearchBox();
```

Comment:
Going to need to have some escaping functions here depending on the type of data. Practically Every time you echo something, it should be escaped.

Check out [`wp_kses_post()`](https://codex.wordpress.org/Function_Reference/wp_kses_post) and the generic `esc_` functions in the codex. [https://codex.wordpress.org/Data_Validation](https://codex.wordpress.org/Data_Validation) is a solid reference guide.